Components/weather.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+
+import {Text, View, ScrollView} from 'react-native';
+
+import {getWeatherFromApi} from '../API/MDBApi';
+import {CityWeather} from './weather.type';
+
+interface State {
+  weather: CityWeather;
+}
+
+interface Props {}
+
+export class Weather extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      weather: {cod: '', message: 0, cnt: 0, list: null, city: null},
+    };
+  }
+
+  componentDidMount = () => {
+    this._loadWeather();
+  };
+
+  _loadWeather = () => {
+    getWeatherFromApi().then(data => {
+      return this.setState({weather: data});
+    });
+  };
+
+  render() {
+    return (
+      <View>
+        <ScrollView>
+          {this.state.weather &&
```

Comment:
What is the difference with 
`this.state.weather?.list?.map`

Reply:
Also, you'd better use a flatlist than a map in a scrollview here.
It's kinda the same result but Flatlist is the optimized component, especially done for that purpose

Reply:
Ok I just realiized that in the mail it's instructed to use Scrollview and not Flatlist.. Strange 😅